test/e2e/plugins_test.go

```diff
@@ -0,0 +1,210 @@
+					})
+
+					t.Run("does not list any other plugin in $PATH", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{knPluginPath2})
+					})
+				})
+
+				t.Run("with --lookup-plugins is true", func(t *testing.T) {
+					lookupPlugins = true
+
+					setup(t)
+					defer teardown(t)
+
+					knFlags := []string{fmt.Sprintf("--plugins-dir=%s", knPluginsDir), "--lookup-plugins=true"}
+					assert.Assert(t, lookupPlugins == true)
+
+					t.Run("list plugin in --plugins-dir", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{knPluginPath2})
+					})
+
+					t.Run("execute plugin in --plugins-dir", func(t *testing.T) {
+						test.runPlugin(t, knFlags, "helloe2e", []string{}, []string{"Hello Knative, I'm a Kn plugin"})
+					})
+
+					t.Run("when other plugins are in $PATH", func(t *testing.T) {
+						var oldPath string
+
+						setupPath := func(t *testing.T) {
+							oldPath = os.Getenv("PATH")
+							err := os.Setenv("PATH", fmt.Sprintf("%s:%s", oldPath, knPluginsDir2))
+							assert.Assert(t, err == nil)
+						}
+
+						tearDownPath := func(t *testing.T) {
+							err = os.Setenv("PATH", oldPath)
+							assert.Assert(t, err == nil)
+						}
+
+						t.Run("list plugin in $PATH", func(t *testing.T) {
+							setupPath(t)
+							defer tearDownPath(t)
+							test.listPlugin(t, knFlags, []string{knPluginPath, knPluginPath2}, []string{})
+						})
+
+						t.Run("execute plugin in $PATH", func(t *testing.T) {
+							setupPath(t)
+							defer tearDownPath(t)
+							test.runPlugin(t, knFlags, "hello2e2e", []string{}, []string{"Hello Knative, I'm a Kn plugin"})
+						})
+					})
+				})
+			})
+		})
+	})
+}
+
+// Private
+
+func (test *e2eTest) createConfig(t *testing.T, configName, configCode, configPath string) (string, error) {
+	configFilename := filepath.Join(configPath, configName)
+	err := ioutil.WriteFile(configFilename, []byte(configCode), FileModeReadWrite)
+	if err != nil {
+		return "", err
+	}
+	return configFilename, nil
+}
+
+func (test *e2eTest) createPlugin(t *testing.T, pluginName, pluginCode, pluginPath string) (string, error) {
+	pluginFilename := filepath.Join(pluginPath, pluginName)
+	err := ioutil.WriteFile(pluginFilename, []byte(pluginCode), FileModeExecutable)
+	if err != nil {
+		return "", err
+	}
+	return pluginFilename, nil
+}
+
+func (test *e2eTest) listPlugin(t *testing.T, knFlags []string, expectedPlugins []string, unexpectedPlugins []string) {
+	knArgs := addArgs([]string{}, knFlags...)
+	knArgs = addArgs(knArgs, []string{"plugin", "list"}...)
+
+	out, err := test.kn.RunWithOpts(knArgs, runOpts{NoNamespace: true})
+	assert.Assert(t, err == nil)
+	assert.Check(t, util.ContainsAll(out, expectedPlugins...))
+	assert.Check(t, util.ContainsNone(out, unexpectedPlugins...))
+}
+
+func (test *e2eTest) runPlugin(t *testing.T, knFlags []string, pluginName string, args []string, expectedOutput []string) {
```

Comment:
`expectedOutput` can just be string here?

Reply:
👍

Reply:
Want to check for multiple lines